spack/packages/dla-future/package.py

```diff
@@ -33,18 +33,23 @@ class DlaFuture(CMakePackage, CudaPackage):
 
     depends_on("umpire~examples")
     depends_on("umpire+cuda~shared", when="+cuda")
+    depends_on("umpire+rocm~shared", when="+rocm")
 
     # https://github.com/eth-cscs/DLA-Future/issues/420
     conflicts("umpire@6:")
 
     depends_on("pika cxxstd=17 +mpi")
-    depends_on("pika@0.4.0")
+    depends_on("pika@0.4.0:")
```

Comment:
Change also this before merge

```suggestion
    depends_on("pika@0.5.0")
```